pkg/kubectl/explain/field_lookup.go

```diff
@@ -78,10 +78,14 @@ func (f *fieldLookup) VisitKind(k *proto.Kind) {
 	if f.SaveLeafSchema(k) {
 		return
 	}
-
-	subSchema, ok := k.Fields[f.Path[0]]
+	var field string = f.Path[0]
+	if len(f.Path) > 1 && f.Path[1] == "" {
+		// validate the last field which can be accidently ended with period
+		field = field + "." + f.Path[1]
+	}
+	subSchema, ok := k.Fields[field]
```

Comment:
If field is `xyz.`, it means that it will never exist, right? So we should probably return early with an error above in the if condition.

Reply:
Agree will all three points, thanks!. I just wanted to continue the flow in case anything change in future but as you said I don't see any harm right now in returning it right away. Let's see what @ahmetb suggest about changes and overall approach?